src/lib/PnP.Framework/Provisioning/Providers/Xml/XmlPnPSchemaBaseSerializer.cs

```diff
@@ -683,8 +683,19 @@ public ProvisioningHierarchy ToProvisioningHierarchy(Stream hierarchy)
 
             XMLPnPSchemaVersion currentSchemaVersion = GetCurrentSchemaVersion();
 
-            var serializers = currentAssembly.GetTypes()
-                // Get all the serializers
+            // Load all loadable types
+            Type[] serializerTypes;
+            try
+            {
+                serializerTypes = currentAssembly.GetTypes();
+            }
+            catch (ReflectionTypeLoadException e)
+            {
+                serializerTypes = e.Types.Where(t => t != null).ToArray();
+            }
+
+            // Get all the serializers
+            var serializers = serializerTypes
```

Comment:
While I'm personally not affiliated with maintaining this repository in any way, I feel this change is a bad one.

The `BuildInFieldId` in the current state is fundamentally broken, and this change only obscures that fact by hiding the error.

Reply:
My pull request fixed two issues. Firstly, the fundamentally wrong strings that break the BuildInFieldId. Secondly, a type loading problem in azure functions. Please keep these two problems apart.

Reply:
I see. I thought those problems were connected.